docs: updates to the seeding documentation

LuisFelipeCoelho opened the pull request:
Some updates to the seeding documentation

tboldagh commented:
The document starts with exclaimer that the intro section is outdated. Is it still needed? The intro is fairly general anyways.

paulgessinger commented on docs/core/seeding.rst:
```diff
@@ -93,77 +101,103 @@ only once, the circle calculation is spread out over the three loops.
-         }
-         // 1/helixradius: (B/sqrt(S2))/2 (we leave everything squared)
-         float iHelixDiameter2 = B2 / S2;
+	for (auto spM : middleSPs) {
+   
+		... // compatibility cuts between SP duplets
+
```
The problems with these code listings is that they have the potential to become outdated very quickly.
Do you see any way you could break this up somehow and describe the sequence in words, while linking to the various functions that are called using the sphinx syntax?

paulgessinger commented:
> The document starts with exclaimer that the intro section is outdated. Is it still needed? The intro is fairly general anyways.

True, with this PR that should probably be removed.

tboldagh commented:
Can't comment o unchanged text. But I have simplifying suggestions about captions:
1 - Sketch of the detector with 3 layers. The interaction region is supposed to be located along the z axis and have size significantly smaller than the radius of the innermost detector layer.
2- The x-y projection of the detector with the charged particle helical track originating from the centre of the detector. Signals left by passage of the track through the detector layers are marked with green crosses. 
3- The r-z projection of the detector with the same charged particle track. The track is depicted with the same colours as on previous figure. 

If you have a chance to touch the plots can they be resized so that on each of them the detector has similar size. 
On the figure 3  I would suggest to mark the radius of the detectror as done in 2.

In general the explanations do not belong to the picture captions but should be part of the text.

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -173,7 +207,7 @@ energy and impact parameters can be estimated.
 redoing calculations. First, the maximum allowed scattering at the configured
 minimum transverse momentum (pT) cut is calculated and scaled by the
-pseudorapidity of the bottomSP-middleSP dupletto get the minimum momentum of
+pseudorapidity of the bottomSP-middleSP duplet to get the minimum momentum of
```
SP from bottom layer SP from middle layer

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -194,46 +228,55 @@ scattering for the triplet as well.
+Lynch & Dahl correction of the Highland equation assuming the lowest
+allowed pT. The parameters of the Highland equation are fully configurable.
 
 The following code block calculates if the triplet forms a nearly straight line
```
calculates -> checks

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -194,46 +228,55 @@ scattering for the triplet as well.
 
 The following code block calculates if the triplet forms a nearly straight line
 in the r/z plane (see :numref:`rz`) as the particle path in the r/z plane is
 unaffected by the magnetic field [#f1]_. This is split in two (may be revised
```
tow parts

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -194,46 +228,55 @@ scattering for the triplet as well.
+			t0 = t + 1;
+		}
+		continue;
+	}
```
As suggested above. Better not to have the code here. Gets out of date quickly. Pseudocode in some form is better.

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -194,46 +228,55 @@ scattering for the triplet as well.
+		continue;
+	}
 
 Now the check for scattering with calculated particle momentum. Momentum is
```
Next/Following check takes into account estimate particle momentum (smaller scattering angle is permitted for higher momentum) and pseudorapidity (larger scattering takes into account amount of the material crosses that takes depends on the angle). (If that is the reason actually.)

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -242,19 +285,34 @@ previous check, as the scattering decreases linearly with particle energy
+		continue;
+	}
 
 The last cut applied in this function is on the so-called impact parameters,
```
transverse impact parameter (or DCA - distance of closest approach) - various experiments refer to it differently

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -263,18 +321,30 @@ compatible with both the current middle SP and current bottom SP.
+		state.ptVec.push_back(pT);
+	}
 
 The bottom SP and middle SP as well as the collection of top SP is passed to
```
Sorry, can't understand this. Maybe simply start with saying that the seeds are filtered after being crated. And instead of details of function calls explain why and how?

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -288,19 +358,20 @@ This function assigns a weight (which should correspond to the likelihood that
 a seed is good) to all seeds and calls the detector specific cuts to apply a
```
calls -> applies detector specific section of seeds based on weights. (not clear what the weight modifications is for)

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -288,19 +358,20 @@ This function assigns a weight (which should correspond to the likelihood that
 
-#. The impact parameter (the higher the distance the worse)
-#. The number of seeds which may belong to the same particle track
+#. The transverse (`d_{0}`) and longitudinal (`z_{0}`) impact parameters (the higher the distance the worse)
```
the worse -> the smaller the weight

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -288,19 +358,20 @@ This function assigns a weight (which should correspond to the likelihood that
-#. The impact parameter (the higher the distance the worse)
-#. The number of seeds which may belong to the same particle track
+#. The transverse (`d_{0}`) and longitudinal (`z_{0}`) impact parameters (the higher the distance the worse)
+#. The number of seeds which may belong to the same particle track (`N_{t}`)
```
how is the weight impacted by this?

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -288,19 +358,20 @@ This function assigns a weight (which should correspond to the likelihood that
+the weight if configured.
 
 The number of seeds only differing in top SP which have similar helix radius
 and the same sign (i.e. the same charge) is used to increase the weight, as it
```
text below introduces "measurement". I suggest to stay with SP. In any case does this cover SPs in the overlapping modules? The logic below is quite hard to follow.

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -313,6 +384,36 @@ configured factor and added to the weight.
 The optional detector specific cuts can use the weight from 1. and 2. and the
```
Until now it seemed that there is only one weight?

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -313,6 +384,36 @@ configured factor and added to the weight.
 The optional detector specific cuts can use the weight from 1. and 2. and the
 three SP to apply a hard cut or change the weight of a seed.
 
+This function also includes a fully configurable seed confirmation step that, when enabled
```
Is this in addition to above? Need to explain the relation of these filters.

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -324,6 +425,12 @@ location. The ordering by location is only done to make sure reimplementations
 
+When a seed is accepted and seed confirmation is enabled, the weight of that seed
+is assigned to each of its SPs. Each SP will hold the weight of the best seed that
+includes that SP. This information is used in the selection of the next seeds:
```
next seeds?

tboldagh commented on docs/core/seeding.rst:
```diff
@@ -324,6 +425,12 @@ location. The ordering by location is only done to make sure reimplementations
+When a seed is accepted and seed confirmation is enabled, the weight of that seed
+is assigned to each of its SPs. Each SP will hold the weight of the best seed that
+includes that SP. This information is used in the selection of the next seeds:
+The seed is kept only if its weight is greater than the weight of at least one of
```
Maybe start by saying that it is some sort of voting scheme that is implemented here. BTW. Greater or equal, right? 
I.e. in the illustrations above there will be only one weight value.

LuisFelipeCoelho commented:
Thanks a lot @tboldagh for your comments! I will update this ASAP

LuisFelipeCoelho commented on docs/core/seeding.rst:
```diff
@@ -11,7 +11,7 @@ find all measurements belonging to a single particle in this direction in order
 particle will not be reconstructed. On the other hand, finding too many seeds
 which either correspond to a particle for which another seed already exists or
-which does not correspond to a particle at all increases the time needed for
+which do not correspond to a particle at all increases the time needed for
```
I thought this referred to "too many seeds" from "finding too many seeds".

AJPfleger replied:
`which either correspond to particles with already existing seeds or
which do not correspond to particles at all increases the time needed for`

I think it should be plural. Maybe rewording the first sentence could make it clearer.
Also, I think `particles` should be plural.

LuisFelipeCoelho replied:
done

LuisFelipeCoelho commented:
@tboldagh @paulgessinger 
I hope that I have addressed all the comments.
I have decided to replace the code block with some equations and further explanation. If you think that I have added too much information or that the pseudo code is still necessary or that I need to add more information, please tell me and I will change it accordingly.

paulgessinger commented on docs/core/seeding.md:
```diff
@@ -0,0 +1,234 @@
+# Track Seeding
```
The rST had an anchor that was linked to:

```suggestion
(seeding_core)=

# Track Seeding
```

paulgessinger commented on docs/core/seeding.md:
```diff
@@ -0,0 +1,234 @@
+
+### The Seed Finder
+
+The The {class}`SeedFilter::createSeedsForGroup` function receives three iterators 
```
```suggestion
The {func}`SeedFilter::createSeedsForGroup` function receives three iterators 
```

paulgessinger commented on docs/core/seeding.md:
```diff
@@ -0,0 +1,234 @@
+
+After creating the potential seeds we apply a seed filter procedure that compares the seeds with other SPs compatible with the seed curvature.
+This process ranks the potential seeds based on certain quality criteria and selects the ones that are more likely to produce high-quality tracks 
+The filter is divided into two functions {class}`SeedFilter::filterSeeds_2SpFixed` and {class}`SeedFilter::filterSeeds_1SpFixed`.
```
```suggestion
The filter is divided into two functions {func}`SeedFilter::filterSeeds_2SpFixed` and {func}`SeedFilter::filterSeeds_1SpFixed`.
```

paulgessinger commented on docs/core/seeding.md:
```diff
@@ -0,0 +1,234 @@
+The $z_0$ term in the weight is not yet fully configurable, but this will change soon.
+:::
+
+The {class}`SeedFilter::filterSeeds_2SpFixed` function also includes a configurable `Acts::SeedConfirmationRangeConfig` seed confirmation step that, when enabled,
```
```suggestion
The {func}`SeedFilter::filterSeeds_2SpFixed` function also includes a configurable {class}`Acts::SeedConfirmationRangeConfig` seed confirmation step that, when enabled,
```

LuisFelipeCoelho commented on docs/core/seeding.md:
```diff
@@ -0,0 +1,234 @@
+detector radius, such that SPs from the same layer cannot be considered compatible. The second function iterates over the seeds with 
+only a common middle layer SP and selects the higher quality combinations.
+
+#### SeedFilter::filterSeeds_2SpFixed
```
Thank you for the comments

LuisFelipeCoelho commented:
not sure why the docs are failing

paulgessinger commented:
I think its

```
/__w/acts/acts/docs/core/seeding.md:190: WARNING: doxygenfunction: Cannot find function "SeedFilter::filterSeeds_2SpFixed" in doxygen xml output for project "Acts" from directory: _build/doxygen-xml
/__w/acts/acts/docs/core/seeding.md:226: WARNING: doxygenfunction: Cannot find function "SeedFilter::filterSeeds_1SpFixed" in doxygen xml output for project "Acts" from directory: _build/doxygen-xml
```

I'm guessing you need to add the `Acts::` namespace before the functions?

AJPfleger commented on docs/core/seeding.md:
```diff
@@ -0,0 +1,240 @@
+
+## Acts Implementation
+
+The seeding implementation in `Core/include/Acts/Seeding/` was written with a focus on parallelism and maintainability and as detector agnostic as possible, only assuming a (near) homogeneous magnetic field with particles originating from the central detector region. Cuts are configurable and can be plugged in as algorithm which is called by the seeding. The seeding works on measurements or “Spac ...
```
```suggestion
The seeding implementation in `Core/include/Acts/Seeding/` was written with a focus on parallelism and maintainability and as detector agnostic as possible, only assuming a (near) homogeneous magnetic field with particles originating from the central detector region. Cuts are configurable and can be plugged in as an algorithm which is called by the seeding. The seeding works on measurements or “SpacePoints” (SP), which need to provide $(x,y,z)$ coordinates with the $z$ axis being along the magnetic field, and $x$ and $y$. For the seeding algorithm to function the particle point of origin has to have a radius smaller than the radius of the detector layer closest to the interaction region. In other words, this seeding algorithm is not suitable for secondary particles originating far from the interaction region.
```

AJPfleger commented on docs/core/seeding.md:
```diff
@@ -0,0 +1,240 @@
+oi
+:::
+
+Assuming the middle layer SP in the origin of the $x/y$ frame, as in {numref}`impactParameter`. 
```
```suggestion
Assuming the middle layer SP is in the origin of the $x/y$ frame, as in {numref}`impactParameter`. 
```

AJPfleger commented on docs/core/seeding.md:
```diff
@@ -0,0 +1,240 @@
+:::
+
+Assuming the middle layer SP in the origin of the $x/y$ frame, as in {numref}`impactParameter`. 
+The distance between the center of the helix and the interaction point (IP) is given by
```
```suggestion
The distance between the centre of the helix and the interaction point (IP) is given by
```
Since we are using Br.Eng. and to match l.78

LuisFelipeCoelho commented:
Thanks a lot @AJPfleger for your comments

LuisFelipeCoelho commented:
Thanks for your revision @benjaminhuth
